pre-signed URL upload for better security

amjedidiah opened the pull request.

amjedidiah committed:
feat(security): implemented pre-signed URL upload approach

BREAKING CHANGE: Replaced direct cloud storage configuration with pre-signed URLs

- Added pre-signed URL endpoint configuration
- Removed cloud storage credentials from frontend
- Deleted cloud storage config files
- Updated upload logic to use pre-signed URLs
- Simplified configuration interface

amjedidiah committed:
fix(upload): resolved CORS issues in pre-signed URL upload

- Added CORS mode to uploadObject fetch request
- Prettier fixes

BREAKING CHANGE: Requires S3 bucket CORS configuration

amjedidiah committed:
fix(app): build and prettier fixes

amjedidiah committed:
chore(app): undoing unnecessary commit changes

amjedidiah committed:
feat(storage): implemented and refined cloud storage SDK integrations

- Added client-side SDK approach for cloud storage providers
- Implemented AWS integration with presigned URLs
- Added credential refresh and SAS token management
- Implemented progress tracking and retry mechanisms
- Fixed type errors and improve error handling
- Removed unused code and align with latest SDK versions

BREAKING CHANGE: Storage configuration requires provider-specific SDK setup

amjedidiah committed:
feat(storage): implemented Azure SDK integrations

- Added client-side SDK approach for Azure provider
- Implemented Azure integration with presigned URLs
- Implemented progress tracking
- Removed unused code and align with latest SDK versions

amjedidiah committed:
chore(app): removed todo from commit

amjedidiah committed:
refactor(app): streamline storageConfig prop values

- Storage config only requires `provider` and `tokenEndpoint` values

amjedidiah committed:
feat(storage): implemented DO SDK integrations

- Added client-side SDK approach for DO provider
- Implemented DO integration with presigned URLs

amjedidiah committed:
feat(app): s3 compliant support

- Unified s3 client-side upload approach for S3 complaint storage providers

amjedidiah committed:
feat(app): merged azure and s3 SDK into one

- preview url for uploaded files

amjedidiah committed:
fix(app): fixed console warning about ETag header

- made ETag header conditional
- removed `previewUrl` from object returned to FE

BSalaeddin reviewed (commented):
### Variable Naming Improvements suggestions

1. **`files`**: Rename to `uploadedFiles` or `selectedFiles` in components and hooks that expose it to developers. This will make its purpose clearer when returned by hooks or used as props in components (e.g., in `UpupUploader.tsx`).

2. **`onChange`**: Rename to `onFileSelectionChange` for props exposed by components to better describe its role in handling file selection updates.

3. **`toBeCompressed`**: Rename to `shouldCompress` for component props to better reflect its boolean nature and improve clarity for developers configuring the component.

4. **`keys`**: Rename to `uploadedFileKeys` if returned by hooks or methods, to clarify that it represents the identifiers of successfully uploaded files.

amjedidiah committed:
refactor(app): variable naming improvements

Renames
- `files` to `selectedFiles`
- `onChange` to `onFilesSelected`
- `onFilesChange` to `onPrepareFiles`
- `toBeCompressed` to `shouldCompress`
- `keys` to `uploadedFileKeys`

MedAmine1212 reviewed (approved):
lgtm

MedAmine1212 commented on src/components/UpupUploader/DropZone.tsx:
```diff
@@ -66,7 +66,7 @@ const DropZone: FC<Props> = ({
                             : // only one file
-                              [acceptedFiles[0]]
+                              [acceptedFiles[0]].filter(Boolean)
                     })
```
Could you explain this please @amjedidiah why did you add the .filter(Boolean) ?

amjedidiah replied:
@MedAmine1212, updated to be more readable and safe.

amjedidiah committed:
fix: safe handling of empty file selection

amjedidiah committed:
chore(app): removed `console.log` statements

amjedidiah committed:
Merge branch 'master' into feat/cloud-storage-sdk-integration
